bigquery2/bigquery_types.bal

```diff
@@ -87,3 +87,20 @@ public type Table record {
     int lastModifiedTime?;
     Field[] fields?;
 };
+
+public type InsertTableData record {
+    string kind = "";
+    InsertError[] insertErrors = [];
+};
+
+public type InsertError record {
+    int index = 0;
+    Error[] errors = [];
+};
+
+public type Error record {
+    string reason = "";
+    string location = "";
+    string debugInfo = "";
+    string message = "";
+};
```

Comment:
Can't we use ballerina builtin error here?

Reply:
From the API we will get response like below for 200 OK. 
{
  "kind": "bigquery#tableDataInsertAllResponse",
  "insertErrors": [
    {
      "index": unsigned integer,
      "errors": [
        {
          "reason": string,
          "location": string,
          "debugInfo": string,
          "message": string
        }
      ]
    }
  ]
}
If we are using ballerina builtin error, Don't we need to pass these details "reason", "location", "debugInfo"  to user?

Reply:
@anupama-pathirage @Biruntha , I also think using error value here is a good thing. But I am not sure current compiler supports this feature OOB. So could you please create an issue to track this improvement, (Checking Possibility of using error values to represent client errors) in Ballerinalang repo? Until then shall we continue with current approach.

Reply:
Created the issue to track

https://github.com/ballerina-platform/ballerina-lang/issues/14297